### DISCUSSION TITLE
Autogen with multiple RAG agents

### DISCUSSION THREAD

**Discussion by egaur** [microsoft/autogen#4398] [Q&A]
I need to create a setup of multiple RAG agents with their own context (which is a DB schema) and a boss agent to call correct RAG agent on the basis of a particular condition. Further from the DB context provided by these RAG agents, another agent need to create a SQL query and run it.. I am stuck in 1st point where the boss agent, userproxy, is not able to call different RAG agents. any leads please.
I am using GPT4o using Azure AI services.

**Comment by ekzhu** [comment]
Could you describe your scenario using code sample?
